docs/sdk/v3/guides/liquidity/04-swap-and-add-liquidity.md

```diff
@@ -0,0 +1,97 @@
+It is based on the [swap-and-add example](https://github.com/Uniswap/examples/tree/main/v3-sdk/swap-and-add-liquidity), found in the Uniswap code examples [repository](https://github.com/Uniswap/examples).
+To run this example, check out the examples's [README](https://github.com/Uniswap/examples/tree/main/v3-sdk/swap-and-add-liquidity) and follow the setup instructions.
+
+:::info
+
+If you need a briefer on the SDK and to learn more about how these guides connect to the examples repository, please visit our [background](./01-background.md) page!
+
+:::
+
+When adding liquidity to a Uniswap v3 pool, you must provide two assets in a particular ratio. In many cases, your contract or the user's wallet hold a different ratio of those two assets. In order to deposit 100% of your assets, you must first swap your assets to the optimal ratio and then add liquidity. 
+
+However, the swap may shift the balance of the pool and thus change the optimal ratio. To avoid that, we can execute this swap-and-add liquidity operation in an atomic fashion, using a router. The inputs to our guide are the **two tokens** that we are pooling for, the **amount** of each token we are pooling for and the Pool **fee**.
+
+The guide will **cover**:
+1. Creating a router instance
+2. Constructing the parameters for the swap-and-add function call
+3. Making the swap-and function call 
+4. Constructing and executing the transaction to swap-and-add liquidity 
+
+At the end of the guide, given the inputs above, we should be able to mint a liquidity position with the press of a button and view the position's id on the UI of the web application. We should also be able to swap-and-add liquidity with the press of a button and see the change reflected in the balance of our tokens.
+
+## Example
+
+### Creating a router instance
+
+The first step is to setup our router, the [`AlphaRouter`](https://github.com/Uniswap/smart-order-router/blob/97c1bb7cb64b22ebf3509acda8de60c0445cf250/src/routers/alpha-router/alpha-router.ts#L333), which is part of the [smart-order-router package](https://www.npmjs.com/package/@uniswap/smart-order-router). The router requires a `chainId` and a `provider` to be initialized. Note that routing is not supported for local forks, so we will use a mainnet provider even when swapping on a local fork:
+
+```js reference title="Creating a router instance" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/a34ecd48c95c075bfbc443af4b4150b481e87b8b/v3-sdk/swap-and-add-liquidity/src/example/Example.tsx#L41
+```
+
+Please note that routing has been covered in greater detail in the [routing guide](../04-routing.md).
+### Constructing the parameters for the swap-and-add function call
+
+
+Having created the router, we now need to construct the parameters required to make a call to its `routeToRatio` function. This is exactly what we want in our example.
```

Comment:
"This is exactly what we want in our example." doesn't really add or say why. maybe:

```suggestion
Having created the router, we now need to construct the parameters required to make a call to its `routeToRatio` function, which will ensure the ratio of currency used matches the pool's required ratio to add our total liquidity.
```